nexus/types/src/quiesce.rs

```diff
@@ -379,12 +427,15 @@ impl SagaQuiesceHandle {
                 init_finished: false,
             })
         } else {
+            let error =
+                error.expect("error is always set when disallowing sagas");
```

Comment:
Nit - you could drop this `.expect()` if we got rid of `okay` entirely and branched based on `error`:

```rust
if let Some(error) = error {
    // ... what's currently in the `else` branch ...
} else {
    // ... what's currently in the `if` branch ...
}